core/src/main/java/io/grpc/internal/AbstractChannelBuilder.java

```diff
@@ -45,7 +48,8 @@
  *
  * @param <BuilderT> The concrete type of this builder.
  */
-public abstract class AbstractChannelBuilder<BuilderT extends AbstractChannelBuilder<BuilderT>> {
+public abstract class AbstractChannelBuilder<BuilderT extends AbstractChannelBuilder<BuilderT>>
+    extends ChannelBuilder<BuilderT> {
```

Comment:
Extending OkHttpChannelBuilder would have the undesirable effect of making it's signatures part of a stable API which is what we're explicitly trying to avoid here.

We could drop the ChannelBuilder interface for the moment  as we don't need to dynamically detect the underlying transport today. That said AbstractChannelBuilder is now an internal type and so is not really part of the public contract.